v2/internal/frontend/runtime/desktop/events.js

```diff
@@ -50,11 +52,13 @@ export const eventListeners = {};
  * @param {string} eventName
  * @param {function} callback
  * @param {number} maxCallbacks
+ * @returns {Listener}
  */
 export function EventsOnMultiple(eventName, callback, maxCallbacks) {
     eventListeners[eventName] = eventListeners[eventName] || [];
-    const thisListener = new Listener(callback, maxCallbacks);
+    const thisListener = new Listener(eventName, callback, maxCallbacks);
     eventListeners[eventName].push(thisListener);
+    return thisListener;
```

Comment:
probably better to return "cancel" callback instead, to avoid exposing internal Listener API

Reply:
This is the question I brought up in my issue. I can see advantages in having a more active style api, and returning the listener itself would facilitate that. Otoh, simply returning the cancel function is the 80% usecase, so I dig that simplicity.

Reply:
Returning a cancel callback was the first thing I thought of too. How easy is it to manage those callbacks though? Are the 2 approaches mutually exclusive?

Reply:
So, I think it would be better to have a design discussion on the issue instead of the PR, but happy to do it here if you're really into that.

In short, if this was my design, I would have a low level Listener interface, and then some higher level functions on top that correspond to the existing api, but with a few tweaks. I would more closely align it with nodejs's eventemitter api with a goal of it being more idiomatic js.

Reply:
And, I'm happy to make the changes to the api to more what I'm describing, just didn't want to actually do that unless people were into it.

Reply:
> Returning a cancel callback was the first thing I thought of too. How easy is it to manage those callbacks though? Are the 2 approaches mutually exclusive?

I think main problem that approach with returning listener object will expose it when it is not intended to be exposed (it seems private interface with limited documentation).